ext/websocket/lib.rs

```diff
@@ -222,38 +223,47 @@ where
   let addr = format!("{domain}:{port}");
   let tcp_socket = TcpStream::connect(addr).await?;
 
-  let socket: MaybeTlsStream<TcpStream> = match uri.scheme_str() {
-    Some("ws") => MaybeTlsStream::Plain(tcp_socket),
+  let socket: NetworkStream = match uri.scheme_str() {
+    Some("ws") => NetworkStream::Tcp(tcp_socket),
     Some("wss") => {
-      let tls_config = create_client_config(
+      let mut tls_config = create_client_config(
         root_cert_store,
         vec![],
         unsafely_ignore_certificate_errors,
         None,
       )?;
-      let tls_connector = TlsConnector::from(Arc::new(tls_config));
-      let dnsname = ServerName::try_from(domain.as_str())
+      // Send http/1.1 for ALPN
+      // https://chromestatus.com/feature/5687059162333184
+      tls_config.alpn_protocols =
+        vec![vec![104, 116, 116, 112, 47, 49, 46, 49]];
+      let server_name = ServerName::try_from(domain.as_str())
         .map_err(|_| invalid_hostname(domain))?;
-      let tls_socket = tls_connector.connect(dnsname, tcp_socket).await?;
-      MaybeTlsStream::Rustls(tls_socket)
+      let mut stream =
+        TlsStream::new_client_side(tcp_socket, tls_config.into(), server_name);
+      stream.handshake().await?;
+      NetworkStream::Tls(stream)
     }
     _ => unreachable!(),
   };
 
-  let client =
-    fastwebsockets::handshake::client(&LocalExecutor, request, socket);
+  let client = crate::handshake::client(request, socket);
```

Comment:
Remove `handshake.rs`. Use `fastwebsockets::handshake::client` and call `into_inner` to get the underlying stream below.

Reply:
`into_inner` was added in fastwebsockets [0.2.6](https://github.com/littledivy/fastwebsockets/commit/3ffe9403724b62cf3eab850ec76fd482eafad081)